### PR TITLE
Export csv improvement

### PR DESCRIPTION
**Pull request by ale7714**

### CODE REVIEW

File: lib/csv_row_model/import/single_model.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          end.first
+          match ? match[1] : nil
+        end
+
+        # @return [Array] header_matchs matchers for the row model
+        def header_matchers
+          @header_matchers ||= begin
+            columns.map do |name, options|
+              matchers = options[:header_matchs] || [name.to_s]
+              Regexp.new(matchers.map(&:downcase).join('|'),Regexp::IGNORECASE)
```
Comment (joel) [inline lib/csv_row_model/import/single_model.rb]: why you downcase here if you precise this regex is unsensitive case?

Comment (ale7714) [inline lib/csv_row_model/import/single_model.rb]: i added Regexp::IGNORECASE, after doing downcasing and forgot to delet. Will update

File: README.md [hunk 2]
```diff
@@ -260,4 +261,40 @@ class ImportFile < CsvRowModel::Import::File
...
+### Export RowModel
+
+Maps each attribute of the `RowModel` to a column of a CSV row.
+
+```ruby
+class ProjectExportRowModel < ProjectRowModel
+  include CsvRowModel::Export
+
+  # Optionally it's possible to override the attribute method, by default it
+  # does source_mode.public_send(attribute)
```
Comment (s12chung) [inline README.md]: "source_mode    l   .public_send(attribute)" :p

File: lib/csv_row_model.rb [hunk 3]
```diff
@@ -31,6 +32,7 @@ class Boolean; end unless defined? Boolean
 
   require 'csv_row_model/export'
   require 'csv_row_model/export_collection/csv'
+  require 'csv_row_model/export_single_model/csv'
```
Comment (s12chung) [inline lib/csv_row_model.rb]: your thoughts on moving `export_collection` to `export/collection` and `export_single_model` to `export/single_model`?

File: lib/csv_row_model/export.rb [hunk 4]
```diff
@@ -7,6 +7,11 @@ module Export
...
 
       self.column_names.each do |column_name|
+
+        # Safe to override
+        #
+        #
+        # @return [String] a string of public_send(column_name) of the CSV model
         define_method(column_name) do
           source_model.public_send(column_name)
         end
```
Comment (s12chung) [inline lib/csv_row_model/export.rb:17]:
``` ruby
class RowModel
  include CsvRowModel::Model

  # calling define_method in included handles this
 column :id

  include CsvRowModel::Export

  # but i bet this doesn't work--you need to override the column method
  column :name
end
```

File: lib/csv_row_model/export.rb [hunk 5]
```diff
@@ -25,6 +30,34 @@ def to_row
...
+
+    # Safe to override
+    #
+    # @return [Array<Array>] an array of arrays, where every represents a row and every row
+    #                        can have strings and row_name (column_name). By default,
+    #                        returns a row_name for every row
+    def rows_temaplate
+      @rows_temaplate ||= self.class.row_names.map{ |row_name| [row_name]}
+    end
+
```
Comment (s12chung) [inline lib/csv_row_model/export.rb]: let's move the changes in this file to a `SingleModel` module to separate this logic from the rest.

File: lib/csv_row_model/model/columns.rb [hunk 6]
```diff
@@ -19,11 +19,13 @@ def to_json
         def column_names
           columns.keys
         end
+        alias_method :row_names, :column_names
```
Comment (s12chung) [inline lib/csv_row_model/model/columns.rb]: let's move these to `SingleModel` where they apply.

**Comment by s12chung** [comment]
feel free to merge. a tough challenge here. let's pair program on this Monday?

**Comment by ale7714** [comment]
Guys thanks for your comments @s12chung @joel, I'll merge and continue improving

**Pull request merged**